src/formatting/items.rs

```diff
@@ -434,6 +434,11 @@ impl<'a> FmtVisitor<'a> {
             FnBraceStyle::None,
         )?;
 
+        // If `result` ends with a comment, then remember to add a newline
+        if last_line_contains_single_line_comment(&result) {
+            result.push_str(&indent.to_string_with_newline(context.config));
+        }
```

Comment:
I may be remembering incorrectly, but I believe the second item in the tuple from `rewrite_fn_base` is a boolean that informs consumers whether a newline is needed, and that internal check includes `last_line_contains_single_line_comment`. 

If so then we should just use that tuple element instead of tossing it and re-running the comment check

Reply:
I already tried that actually. But `force_new_line` is `true` in other cases, that would break existing test cases

For instance, if I replaced it with `if force_new_line` as returned by the second tuple item from `rewrite_fn_base`, then e.g. the [`max_width.rs`] test case would break, and result in:

[`max_width.rs`]: tests/source/configs/space_before_fn_sig_paren/max_width.rs

```rust
trait Story {
    fn swap_context<T: 'static + Context + Send + Sync>(&mut self, context: T) -> Option<Box<Context + Send + Sync>>
    ;
}
```

Instead of remain as:

```rust
trait Story {
    fn swap_context<T: 'static + Context + Send + Sync>(&mut self, context: T) -> Option<Box<Context + Send + Sync>>;
}
```

Reply:
Gotcha, makes sense. If the current `rewrite_fn_base` newline_force check includes this same comment check, could we refactor the returned tuple to include a third element that also has result of that comment check, and then use that here (instead of the current second item)? Trying to avoid running the same comment check twice if possible

If that's not feasible then I think this will be fine, as it's not exactly expensive

Reply:
I did consider that as well, but I was unsure if changing the signature of `rewrite_fn_base` was acceptable. I've just refactored it, and introduced a third item in the tuple.

Reply:
> I've just refactored it, and introduced a third item in the tuple.

Great, thanks!

> I was unsure if changing the signature of rewrite_fn_base was acceptable.

Absolutely acceptable. We do publish rustfmt as a lib so there's some strict constraints on the public portions of the API, but for anything internal we can and do change whenever needed